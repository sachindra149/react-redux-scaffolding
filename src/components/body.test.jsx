import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { render, shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './body.jsx';


configure({adapter: new Adapter()});

describe("<App />", () => {

	it("Component should be defined", () => {
		expect(App).toBeDefined();
	});

	it("Should simply render the component", () => {
		const wrapper = render(<App />);
		expect(wrapper);
	});
	
	it("Should match the snapshot", () => {
		const tree = renderer.create(<App />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Testing Shallow and Mount Rendering", () => {
		const shallowWrapper = shallow(<App />);
		const mountWrapper = mount(<App />);
		expect(shallowWrapper.containsMatchingElement(<h1>React Redux Testing App</h1>)).toEqual(true);
		expect(mountWrapper.containsMatchingElement(<h1>React Redux Testing App</h1>)).toEqual(true);
	});

	it("Should render a loader initially", () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find('table').length).toEqual(0);
		expect(wrapper.find('.loading').length).toEqual(1);
	});

});
