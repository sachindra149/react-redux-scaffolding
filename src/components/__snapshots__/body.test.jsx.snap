// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should match the snapshot 1`] = `
Array [
  <h1
    className="fullWidth"
  >
    React Redux Testing App
  </h1>,
  <h3
    className="fullWidth"
  >
    This is a basic scaffolding for app using React And Redux
  </h3>,
  <div
    className="fullWidth"
  >
    <div
      className="loading"
    >
      <div
        className="lds-dual-ring"
      />
    </div>
  </div>,
]
`;
